### PR TITLE
fix for recursive file src

### PR DESCRIPTION
**Pull request by ghost**
Hi there, I've forked your repo to fix a bug/add a feature to allow recursive src's to be copied that are relative to the gulp file.

### CODE REVIEW

**Comment by popomore** [comment]
Why change the indent?

**Comment by ghost** [comment]
No reason I have auto formatting on as default n my editor

**Comment by popomore** [comment]
Can you revert it to 2 space, you can change the setting to detect the indent automatically

**Comment by ghost** [comment]
Yes no problem, will do!

**Comment by ghost** [comment]
Ok I've set the indent on my changes to 2

File: index.js [hunk 1]
```diff
@@ -20,31 +20,38 @@ module.exports = function(options) {
...
+  return through.obj(function(file, enc, callback) {
+      if (file.isStream()) {
+        return callback(new PluginError('gulp-scp2', 'Streaming not supported.'));
       }
 
-      client.write({
-        destination: path,
-        content: file.contents
-      }, function(err) {
+      var filePath = pathLib.relative(process.cwd(), file.path);
```
Comment (popomore) [inline index.js:28]: Why not use `file.relative`?  the path of the file is not determined by where you run the command.

**Comment by popomore** [comment]
I know what you want, you can give the cwd to `gulp.src`, actually the default is `process.cwd()`

```
gulp.src('**/*.js', {cwd: process.cwd()})
  .pipe(scp({
    host: 'localhost',
    username: 'username',
    password: 'password',
    dest: '/home/username/'
  }));
```

https://github.com/gulpjs/gulp/blob/master/docs/API.md#optionscwd

File: index.js (hunk 1 above)
Comment (ghost) [inline index.js:28]: The problem I was trying to solve was to regenerate on the server (destination) the folder structure, if I had used an absolute path this wouldn't have worked.

Comment (popomore) [inline index.js:28]: file.relative is not absolute path, and is base on file.cwd.

Can you show me your code.

Comment (ghost) [inline index.js:28]: gulp.task('deployfull', function() {
    return gulp.src(['_.js', '_.json', './public/**/*.{js,html,css}'])
        .pipe(debug())
        .pipe(scp({
            host: '192.168.0.108',
            username: '**_',
            password: '_**',
            path: '/home/pi/',
            showProgress: true
        }))
        .on('error', function(err) {
            console.log(err);
        });

});

Comment (popomore) [inline index.js:28]: You can use glob for that, 

```
gulp.src(['**.js', '**.json']);
```

`**` stand for everything containing `/`

Comment (ghost) [inline index.js:28]: Ok forget the pull request your code does not generate folder structure in the destination. I will write my own.

**Comment by phillipgreenii** [comment]
Is this PR dead or will it be merged to support recursive copying?

**Comment by popomore** [comment]
@phillipgreenii already supported, you can see comment above. you can use glob, and it will generator folder in remote server.

**Pull request closed without merging**